_posts/back-to-basics-boundaries.md

```diff
@@ -0,0 +1,75 @@
+
+> For a detailed explanation of what "logical" means in this context, look at [Let's get logical! On logical and physical architectural views](https://milestone.topics.it/2022/01/25/lets-get-logical.html)
+
+### Service Boundaries
+
+As a consequence of services' existence, a service boundary is a _virtual_ fence that protects services from being _contaminated_ by components not belonging to them. It also prevents leakages. Components belonging to a service won't leak into other services.
+
+## More in-depth articles
+
+I discussed services, components, and coupling from various perspectives in the past. Here is a non-exhaustive list of articles covering the mentioned topics:
+
+- In [Not all changes are born equal](https://milestone.topics.it/2021/03/10/not-all-changes-are-born-equal.html), I look at what coupling is and how it affects or doesn't various parts of the system based on the type of changes we're facing.
+- [There is no such thing as orchestration](https://milestone.topics.it/2021/07/08/no-orchstration.html) looks at orchestration and the inherited coupling it brings to the table.
+- The focus is on cache misuse leading to cumbersome implementations in [Own the cache!](https://milestone.topics.it/2021/07/15/own-the-cache.html).
+- [Update me, please](https://milestone.topics.it/2021/08/03/update-me-please.html) takes an entirely different point of view and looks at a notification system and what's its relationship with coupling.
+- [Distributed systems evolution challenges](https://milestone.topics.it/2022/06/11/distributed-systems-evolution-challenges.html) is a short series of articles detailing the challenges and options to evolve a live distributed system and the role played by coupling.
+- [How many (micro)services do I need?](https://milestone.topics.it/2023/03/15/how-many-services.html) discusses what happens when we go crazy and end up with too many autonomous and business components and too many services.
+- Last but not least, [Autonomy probably doesn't mean what you think it means](https://milestone.topics.it/2022/09/05/autonomy.html) discusses the autonomy concept as the other side of the medal of coupling.
+
+## Conclusion
+
+When designing software systems, we have many tools at our disposal. They range from practical ones, like programming languages or integrated development environments (IDE), to more meta types of tools, like the architectural constructs discussed in this article. All of them are critical for success. Architectural constructs like autonomous and business components, or services and service boundaries, are particularly helpful in defining virtual fences to avoid concepts belonging to different business concerns from leaking and melting into each other, creating a spaghetti mess.
```

Comment:
```suggestion
When designing software systems, we have many tools at our disposal. They range from practical ones, like programming languages or integrated development environments (IDEs), to more meta types of tools, like the architectural constructs discussed in this article. All of them are critical for success. Architectural constructs like autonomous and business components, or services and service boundaries, are particularly helpful in defining virtual fences to avoid concepts belonging to different business concerns from leaking and melting into each other, creating a spaghetti mess.
```